ably/websocket.go

```diff
@@ -1,55 +1,94 @@
 package ably
 
 import (
+	"context"
+	"encoding/json"
 	"errors"
-	"net"
+	"net/http"
 	"net/url"
 	"time"
 
 	"github.com/ably/ably-go/ably/internal/ablyutil"
-	"golang.org/x/net/websocket"
+	"nhooyr.io/websocket"
+)
+
+type proto int
+
+const (
+	jsonProto proto = iota
+	msgpackProto
 )
 
 type websocketConn struct {
 	conn  *websocket.Conn
-	codec websocket.Codec
+	proto proto
 }
 
 func (ws *websocketConn) Send(msg *protocolMessage) error {
-	return ws.codec.Send(ws.conn, msg)
+	switch ws.proto {
+	case jsonProto:
+		p, err := json.Marshal(msg)
+		if err != nil {
+			return err
+		}
+		return ws.conn.Write(context.Background(), websocket.MessageText, p)
+	case msgpackProto:
+		p, err := ablyutil.MarshalMsgpack(msg)
+		if err != nil {
+			return err
+		}
+		return ws.conn.Write(context.Background(), websocket.MessageBinary, p)
+	}
+	return nil
 }
 
 func (ws *websocketConn) Receive(deadline time.Time) (*protocolMessage, error) {
 	msg := &protocolMessage{}
+	var (
+		ctx    context.Context
+		cancel context.CancelFunc
+	)
 	if !deadline.IsZero() {
```

Comment:
Just a small readability suggestion here. Can we make the if statement a positive condition by losing the `!` and switching the statements around.
e.g.
```
	if deadline.IsZero() {
		ctx = context.Background()
	} else {
		ctx, cancel = context.WithDeadline(context.Background(), deadline)
		defer cancel()
	}
```